crates/plugins/src/install.rs

```diff
@@ -48,38 +117,55 @@ impl PluginInstaller {
             return Err(anyhow!("This plugin is not supported on this OS"));
         };
         // TODO: Add logic for architecture as well
-        let plugin_package = plugin
+        let plugin_package = plugin_manifest
             .packages
             .iter()
             .find(|p| p.os == os)
             .ok_or_else(|| anyhow!("This plugin does not support this OS"))?;
         let target_url = plugin_package.url.to_owned();
-        // TODO: Ask for User confirmation
-        if !Prompter::new(
-            &plugin.name,
-            &plugin.license,
-            self.url.to_owned(),
-            &target_url,
-        )?
-        .run()?
-        {
-            // User has requested to not install package, returning early
-            println!("Plugin {} will not be installed", plugin.name);
-            return Ok(());
+
+        // Ask for user confirmation if not overridden wiht CLI option
+        if !self.skip_accept_install {
+            if !Prompter::new(
+                &plugin_manifest.name,
+                &plugin_manifest.license,
+                self.url.to_owned(),
+                &target_url,
+            )?
+            .run()?
+            {
+                // User has requested to not install package, returning early
+                println!("Plugin {} will not be installed", plugin_manifest.name);
+                return Ok(());
+            }
         }
-        // TODO: Handle licensing of plugins
 
         let temp_dir = tempdir()?;
         let plugin_file_name = self.download_plugin(&temp_dir, &target_url).await?;
         self.verify_checksum(&plugin_file_name, &plugin_package.sha256)?;
 
-        self.untar_plugin(&plugin_file_name)?;
+        self.untar_plugin(&plugin_file_name, &plugin_manifest.name)?;
         // Save manifest to installed plugins directory
-        self.add_to_manifest_dir(&plugin)?;
+        self.add_to_manifest_dir(&plugin_manifest)?;
+        log::info!("Plugin: {} installed successfully", self.name);
         Ok(())
     }
 
-    fn untar_plugin(&self, plugin_file_name: &PathBuf) -> Result<()> {
+    fn get_latest_plugin_repo(&self) -> Result<()> {
+        Repository::clone(&self.url.to_string(), self.plugins_dir.join(".plugins"))?;
+        Ok(())
+    }
+
+    fn update_plugins_repository(&self) -> Result<(), git2::Error> {
+        let remote_name = "origin";
+        let remote_branch = "main";
+        let repo = Repository::open(self.plugins_dir.join(".plugins"))?;
```

Comment:
This `.plugins` dir should be a constant

Reply:
agree